internal/incoming/odfi/processor.go

```diff
@@ -81,36 +82,16 @@ func (pcs Processors) HandleAll(file File) error {
 	return el
 }
 
-func ProcessFiles(dl *downloadedFiles, auditSaver *AuditSaver, fileProcessors Processors) error {
+func ProcessFiles(dl *downloadedFiles, auditSaver *AuditSaver, fileProcessors Processors, agent upload.Agent) error {
 	var el base.ErrorList
-	entries, err := os.ReadDir(dl.dir)
-	if err != nil {
-		return fmt.Errorf("reading %s: %v", dl.dir, err)
-	}
-	for i := range entries {
-		where := filepath.Join(dl.dir, entries[i].Name())
 
-		info, err := entries[i].Info()
-		if err != nil {
-			el.Add(fmt.Errorf("processFiles: %v", err))
-			continue
-		}
-
-		if info.Mode().IsDir() {
-			err = processDir(where, auditSaver, fileProcessors)
-			if err != nil {
-				el.Add(fmt.Errorf("processDir %s: %v", info, err))
-				continue
-			}
-		}
-		if info.Mode().IsRegular() {
-			err = processFile(where, auditSaver, fileProcessors)
-			if err != nil {
-				el.Add(fmt.Errorf("processfile - %s: %v", info, err))
-				continue
-			}
+	for _, processingPath := range []string{agent.InboundPath(), agent.ReconciliationPath(), agent.ReturnPath()} {
```

Comment:
Took inspiration from some of the other utilities that are operating on these files such as `Cleanup` and `CleanupEmptyFiles` which directly use the agent's path attributes `InboundPath()`, etc.

This allows the `xPath()` to be arbitrary "deep" but not need to rely on recursion in the `processDir` directory which had raised some concerns and is admittedly a bit less direct.